frb_macros/src/lib.rs

```diff
@@ -10,25 +10,50 @@ use proc_macro::*;
 // frb-coverage:ignore-start
 #[proc_macro_attribute]
 pub fn frb(attribute: TokenStream, item: TokenStream) -> TokenStream {
-    let item_str = item.to_string();
-    let mut new_str = String::with_capacity(item_str.len());
-    let mut last_end = 0;
-    while let Some(start) = item_str[last_end..].find("#[frb(") {
-        let start = start + last_end;
-        let end = item_str[start..].find(")]").unwrap() + start + 2;
-        new_str.push_str(&item_str[last_end..start]);
-        new_str.push_str("\n/// frb_marker: ");
-        new_str.push_str(&item_str[start..end]);
-        new_str.push('\n');
-        last_end = end;
-    }
-    new_str.push_str(&item_str[last_end..]);
-    let item: TokenStream = new_str.parse().unwrap();
+    let mut output = format_frb_attribute(format!("#[frb({attribute})]"));
+    let item = strip_frb_attr(item);
+    output.extend(item);
+    output
+}
+
+fn strip_frb_attr(item: TokenStream) -> TokenStream {
+    item.into_iter()
+        .scan(None, |pound, tok| {
+            use TokenTree as T;
```

Comment:
nit: `T` seems like generic argument, so maybe ``use TokenTree::*` or `as TT` may be better. But it depends on you and not a big problem